subprojects/model-core/src/main/java/org/gradle/internal/reflect/DefaultTypeValidationContext.java

```diff
@@ -25,26 +25,26 @@
 import javax.annotation.Nullable;
 
 public class DefaultTypeValidationContext extends ProblemRecordingTypeValidationContext {
-    private final boolean cacheable;
+    private final boolean reportCacheabilityProblems;
     private final ImmutableMap.Builder<String, Severity> problems = ImmutableMap.builder();
 
     public static DefaultTypeValidationContext withRootType(DocumentationRegistry documentationRegistry, Class<?> rootType, boolean cacheable) {
         return new DefaultTypeValidationContext(documentationRegistry, rootType, cacheable);
     }
 
-    public static DefaultTypeValidationContext withoutRootType(DocumentationRegistry documentationRegistry, boolean cacheable) {
-        return new DefaultTypeValidationContext(documentationRegistry, null, cacheable);
+    public static DefaultTypeValidationContext withoutRootType(DocumentationRegistry documentationRegistry, boolean reportCacheabilityProblems) {
+        return new DefaultTypeValidationContext(documentationRegistry, null, reportCacheabilityProblems);
     }
 
-    private DefaultTypeValidationContext(DocumentationRegistry documentationRegistry, @Nullable Class<?> rootType, boolean cacheable) {
+    private DefaultTypeValidationContext(DocumentationRegistry documentationRegistry, @Nullable Class<?> rootType, boolean reportCacheabilityProblems) {
         super(documentationRegistry, rootType, null);
-        this.cacheable = cacheable;
+        this.reportCacheabilityProblems = reportCacheabilityProblems;
     }
 
     @Override
     protected void recordProblem(TypeValidationProblem problem) {
         boolean cacheableProblemOnly = problem.isCacheabilityProblemOnly();
```

Comment:
I think this should be renamed to `onlyAffectsCacheableWork` (both variable and getter), the current name is confusing.